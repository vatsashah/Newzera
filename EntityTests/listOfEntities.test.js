// Importing required library
import React from 'react'
import { shallow, mount } from 'enzyme'

// Importing component to be tested
import EntityList from '../../../Components/Entity/listOfEntities'

describe('Entity List should render without any error', () => {
  it('Entity List should render correctly', () => {
    const component = shallow(<EntityList />)
    expect(component).toMatchSnapshot()
  })
  it('Entities in EntityList should have same number of entity names and entity images', () => {
    const wrapper = mount(<EntityList />).props()
    const numberOfEntityNames = wrapper.entityName.length
    const numberOfEntityImages = wrapper.entityImg.length
    expect(numberOfEntityImages).toEqual(numberOfEntityNames)
  })
})
