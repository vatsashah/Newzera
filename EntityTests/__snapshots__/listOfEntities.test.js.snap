// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity List should render without any error Entity List should render correctly 1`] = `
<div>
  <div
    className="EntityListCard"
    key="0"
    style={
      Object {
        "left": "0vw",
        "top": "0vh",
      }
    }
  >
    <div>
      <img
        alt="Entity"
        className="EntityListImage"
        src="India.png"
      />
    </div>
    <div>
      <p
        className="EntityListName"
      >
        India
      </p>
    </div>
    <div>
      <p
        className="EntityListNumFollowing"
      >
        52K Following
      </p>
    </div>
    <div
      className="PointerWhenOnHover EntityInListIsfollowing"
    >
      <RectangleButton
        bordered={true}
        disabled={false}
        filled={false}
        fontSize={18}
        height={35}
        radius={38.5}
        title="Follow"
        width={160}
      />
    </div>
  </div>
  <div
    className="EntityListCard"
    key="1"
    style={
      Object {
        "left": "0vw",
        "top": "16.85185185185185vh",
      }
    }
  >
    <div>
      <img
        alt="Entity"
        className="EntityListImage"
        src="usaSq.jpg"
      />
    </div>
    <div>
      <p
        className="EntityListName"
      >
        United States of America
      </p>
    </div>
    <div>
      <p
        className="EntityListNumFollowing"
      >
        52K Following
      </p>
    </div>
    <div
      className="PointerWhenOnHover EntityInListIsfollowing"
    >
      <RectangleButton
        bordered={false}
        disabled={false}
        filled={false}
        fontSize={18}
        height={35}
        radius={38.5}
        title="Following"
        width={160}
      />
    </div>
  </div>
</div>
`;
